src/stores/thumbnails.js

```diff
@@ -0,0 +1,144 @@
+import { defineStore } from 'pinia'
+import { LRUCache } from 'lru-cache'
+import { deleteCachedImageFromBasename, getImageFromBasename, getAllCacheKeys } from '@/utils/imageLoader'
+
+
+const cacheOptionsSmall = {
+  max: 5000,
+  dispose: (value, key, reason) => {
+    if (reason != 'set'){
+      deleteCachedImageFromBasename('small', key)
+    }
+  },
+  noUpdateTTL: true,
+  // cache files for one week
+  ttl: 1000 * 60 * 60 * 24 * 7,
+  ttlResolution: 100*60*60,
+  allowStale: false,
+  updateAgeOnGet: true,
+  updateAgeOnHas: true,
+}
+
+const cacheOptionsLarge = {
+  max: 1000,
+  dispose: (value, key, reason) => {
+    if (reason != 'set'){
+      deleteCachedImageFromBasename('large', key)
+    }
+  },
+  noUpdateTTL: true,
+  // cache files for one week
+  ttl: 1000 * 60 * 60 * 24 * 7,
+  ttlResolution: 100*60*60,
+  allowStale: false,
+  updateAgeOnGet: true,
+  updateAgeOnHas: true,
+}
+
+export const useThumbnailsStore = defineStore('thumbnails', {
+  state() {
+    return {
+      archives: [],
+      smallThumbnailsCache: new LRUCache(cacheOptionsSmall),
+      largeThumbnailsCache: new LRUCache(cacheOptionsLarge),
+    }
+  },
+  persist: {
+    key: 'thumbnails',
+    serializer: {
+      serialize: value => {
+        let serializedValue = {
+          archives: value.archives,
+          smallThumbnailsCache: value.smallThumbnailsCache.dump(),
+          largeThumbnailsCache: value.largeThumbnailsCache.dump()
+        }
+        return JSON.stringify(serializedValue)
+      },
+      deserialize: value => {
+        let tempValue = JSON.parse(value)
+        var smallThumbnailsCache2 = new LRUCache(cacheOptionsSmall)
+        smallThumbnailsCache2.load(tempValue.smallThumbnailsCache)
+        // Clear existing keys URL since they will be invalid on reload
+        for (const key of [...smallThumbnailsCache2.keys()]) {
+          const value = smallThumbnailsCache2.peek(key)
+          URL.revokeObjectURL(value)
+          smallThumbnailsCache2.set(key, '')
+        }
+        tempValue.smallThumbnailsCache = smallThumbnailsCache2
+        var largeThumbnailsCache2 = new LRUCache(cacheOptionsLarge)
+        largeThumbnailsCache2.load(tempValue.largeThumbnailsCache)
+        // Clear existing keys URL since they will be invalid on reload
+        for (const key of [...largeThumbnailsCache2.keys()]) {
+          const value = largeThumbnailsCache2.peek(key)
+          URL.revokeObjectURL(value)
+          largeThumbnailsCache2.set(key, '')
+        }
+        tempValue.largeThumbnailsCache = largeThumbnailsCache2
+        return (tempValue)
```

Comment:
Special code is needed to serialize/deserialize the LRUCache because only simple data can be stored in browser localStorage by the pinia persistedstate plugin. We also clear out the old expired generated blob URLs on reload here, to prepare for regenerating them in the afterRestore callback